core/src/main/java/com/linecorp/armeria/client/HttpRequestSubscriber.java

```diff
@@ -192,11 +191,11 @@ private void writeFirstHeader() {
         if (request.isEmpty()) {
             state = State.DONE;
             write0(firstHeaders, true, true,
-                   reqCtx.additionalRequestHeaders(), requestSessionProtocol, authorityHeader, userAgentHeader);
+                   reqCtx.additionalRequestHeaders(), requestSessionProtocol, authorityHeader);
```

Comment:
You can also remove the `authorityHeader`, because it can be derived from `firstHeaders`, `reqCtx.additionalRequestHeaders()` and `channel.remoteAddress()`. Note that `HttpObjectEncoder` and its subtypes always have a reference to the current `channel`.

Reply:
Thanks for education 😀